scaffoldmaker/meshtypes/meshtype_3d_solidsphere1.py

```diff
@@ -0,0 +1,510 @@
+"""
+Generates a 3-D unit solid sphere mesh with variable numbers of elements
+around, up the central axis, and radially.
+"""
+
+from __future__ import division
+import math
+from scaffoldmaker.utils.eftfactory_tricubichermite import eftfactory_tricubichermite
+from scaffoldmaker.utils.interpolation import *
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.zinc_utils import *
+from opencmiss.zinc.element import Element, Elementbasis, Elementfieldtemplate
+from opencmiss.zinc.field import Field
+from opencmiss.zinc.node import Node
+
+class MeshType_3d_solidsphere1:
+    '''
+    classdocs
+    '''
+    @staticmethod
+    def getName():
+        return '3D Solid Sphere 1'
+
+    @staticmethod
+    def getDefaultOptions():
+        return {
+            'Number of elements around' : 8,
+            'Number of elements up' : 8,
+            'Number of elements radial' : 1,
+            'Diameter' : 1,
+            'Use cross derivatives' : False,
+            'Refine' : False,
+            'Refine number of elements around' : 1,
+            'Refine number of elements up' : 1,
+            'Refine number of elements radial' : 1
+        }
+
+    @staticmethod
+    def getOrderedOptionNames():
+        return [
+            'Number of elements around',
+            'Number of elements up',
+            'Number of elements radial',
+            'Diameter',            
+            'Use cross derivatives',
+            'Refine',
+            'Refine number of elements around',
+            'Refine number of elements up',
+            'Refine number of elements radial'
+        ]
+
+    @staticmethod
+    def checkOptions(options):
+        for key in [
+            'Number of elements radial',
+            'Refine number of elements around',
+            'Refine number of elements up',
+            'Refine number of elements radial']:
+            if options[key] < 1:
+                options[key] = 1
+        if options['Number of elements up'] < 4:
+            options['Number of elements up'] = 4
+        if options['Number of elements around'] < 4:
+            options['Number of elements around'] = 4
+        if options['Diameter'] < 0:
+            options['Diameter'] = 1
+
+    @staticmethod
+    def generateBaseMesh(region, options):
```

Comment:
Change signature to:
    @classmethod
    def generateBaseMesh(cls, region, options):
For consistency with new scaffolds.